### PR TITLE
bots: Add tests for whitelist and don't auto read it

### PR DESCRIPTION
**Pull request by stefwalter**
There's no need to automatically read the github whitelist.
In addition we should have tests for loading and parsing it.

### CODE REVIEW

File: bots/github/test-github [hunk 1]
```diff
@@ -117,5 +117,28 @@ class TestGitHub(unittest.TestCase):
...
+    def testLoad(self):
+        filename = os.path.join(self.temp, "whitelist")
+        with open(filename, "w") as f:
+            f.write("   one   \n   two\nthree\n\n")
+        whitelist = github.whitelist(filename)
+        self.assertIn("one", whitelist)
+        self.assertIn("two", whitelist)
+        self.assertIn("three", whitelist)
+        self.assertNotIn("four", whitelist)
+        self.assertNotIn("", whitelist)
```
Comment (martinpitt) [inline bots/github/test-github]: I'd make this both simpler and more precise with `self.assertEqual(set(github.whitelist(filename)), set(["one", "two", "three"]))`. The first `set()` makes the test robust against possible implementation changes (like returning a list or a generator).

File: bots/github/test-github [hunk 2]
```diff
@@ -117,5 +117,28 @@ class TestGitHub(unittest.TestCase):
...
+
+    def testNotPresent(self):
+        filename = os.path.join(self.temp, "whitelist")
+        whitelist = github.whitelist(filename)
+        self.assertEqual(len(whitelist), 0)
+
```
Comment (martinpitt) [inline bots/github/test-github]: I'd add a new test that calls `github.whitelist()` with the default argument, and assert that the length is > 0. That will ensure that reading the default whitelist does at least something sensible, the whitelist is at the right place, and it doesn't crash.

**Comment by martinpitt** [comment]
It seems we don't actually run `test-github` anywhere in CI?

**Comment by stefwalter** [comment]
> It seems we don't actually run test-github anywhere in CI?

It is run via ```tools/test-bots```

File: bots/github/test-github (hunk 2 above)
Comment (stefwalter) [inline bots/github/test-github]: Done

File: bots/github/test-github (hunk 1 above)
Comment (stefwalter) [inline bots/github/test-github]: Done

File: bots/github/test-github [hunk 3]
```diff
@@ -117,5 +117,30 @@ class TestGitHub(unittest.TestCase):
...
+        shutil.rmtree(self.temp)
+
+    def testLoad(self):
+        filename = os.path.join(self.temp, "whitelist")
+        with open(filename, "w") as f:
+            f.write("   one   \n   two\nthree\n\n")
+        whitelist = github.whitelist(filename)
+        self.assertEqual(set(github.whitelist(filename)), set(["one", "two", "three"]))
+        self.assertNotIn("four", whitelist)
+        self.assertNotIn("", whitelist)
```
Comment (martinpitt) [inline bots/github/test-github]: The last two are now redundant.

Thanks for the fixes! Looks good now.

**Comment by stefwalter** [comment]
This is blocking testing of stuff not based on or rebased onto master, due to:

```
[stef@dragon cockpit]$ PRIORITY=0008 TEST_NAME='pull-6674-20170524-015353' TEST_REVISION='c3a42695d4e36bdc05676013a9bf2fbf830924e6' bots/tests-invoke --rebase='rhel-7.4' 'verify/fedora-atomic' 'pull/6674/head'
remote: Counting objects: 7, done.
remote: Total 7 (delta 2), reused 2 (delta 2), pack-reused 5
Unpacking objects: 100% (7/7), done.
From github.com:cockpit-project/cockpit
 * branch            refs/pull/6674/head -> FETCH_HEAD
HEAD is now at c3a4269... test: Disable all third-party repos in semaphore
From github.com:cockpit-project/cockpit
 * branch            rhel-7.4   -> FETCH_HEAD
Traceback (most recent call last):
  File "bots/tests-invoke", line 305, in <module>
  File "bots/tests-invoke", line 58, in main
  File "bots/tests-invoke", line 251, in run
  File "bots/tests-invoke", line 79, in start_publishing
  File "/data/src/cockpit/bots/github/__init__.py", line 122, in __init__
  File "/data/src/cockpit/bots/github/__init__.py", line 88, in read_whitelist
IOError: [Errno 2] No such file or directory: '/data/src/cockpit/bots/github/../github/whitelist'
```

**Pull request merged**